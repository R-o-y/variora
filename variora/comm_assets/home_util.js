import { notification } from 'antd'


const MAX_DOCUMENT_UPLOAD_SIZE = 5 * 1024 * 1024

function validateDocumentTitle(title) {
  const INVALID_DOCUMENT_NAME_PATTERN = /[]/gm
  if (title == undefined || title == '') {
    notification['warning']({
      message: 'Document title cannot be empty',
      duration: 1.8,
    })
    return false
  }
  if (title.match(INVALID_DOCUMENT_NAME_PATTERN) != null) {
    notification['warning']({
      message: 'The document name contains invalid character',
      description: 'The special characters you can include in your document name are "-|&_.():[]@<>"',
      duration: 6,
    })
    return false
  }
  return true
}

function validateDocumentSize(file) {
  if (file.size > MAX_DOCUMENT_UPLOAD_SIZE) {
    notification['info']({
      message: 'Document size should be less than 3 MB',
      duration: 2.8,
    })
    return false
  }
  return true
}

export { validateDocumentTitle, validateDocumentSize }
