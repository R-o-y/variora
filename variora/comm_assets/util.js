function getUrlFormat(urlBase, paraDic) {
  var url = urlBase + '?'
  for (var key in paraDic)
    url = url + key + '=' + paraDic[key] + '&'
  return url
}


// this is the helper function for getting csrf_token
function getCookie(name) {
  var cookieValue = null
  if (document.cookie && document.cookie != '') {
    var cookies = document.cookie.split(';')
    for (var i = 0; i < cookies.length; i++) {
      var cookie = cookies[i].trim()
      // Does this cookie string begin with the name we want?
      if (cookie.substring(0, name.length + 1) == (name + '=')) {
        cookieValue = decodeURIComponent(cookie.substring(name.length + 1))
        break
      }
    }
  }
  return cookieValue
}


// /**
//  * every 8ms, check whether the specified img finsih being loaded, if so, call the specified callback function
//  * @param  {imgDomElement} img
//  * @param  {function} callback
//  * @return {undefined}
//  */
// function imgLoad(img, callback) {
//   var timer = setInterval(function() {
//     if (img.complete) {
//       callback(img)
//       clearInterval(timer)
//     }
//   }, 8)
// }


function hexToRgb(hex) {
  // Expand shorthand form (e.g. "03F") to full form (e.g. "0033FF")
  var shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])$/i
  hex = hex.replace(shorthandRegex, function(m, r, g, b) {
    return r + r + g + g + b + b
  })

  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex)
  return result ? {
    r: parseInt(result[1], 16),
    g: parseInt(result[2], 16),
    b: parseInt(result[3], 16)
  } : null
}


function formatOpenCoterieDocumentUrl(document, coterieId) {
  return '/coteries/' + coterieId + '/documents/' + document.slug + '/' + document.title.replace(/\s/g, '-')
  // return getUrlFormat('/coterie/display_coteriefile_viewer_page', {
  //   'coterie_id': coterieId,
  //   'document_id': documentId,
  //   'csrfmiddlewaretoken': getCookie('csrftoken'),
  // })
}


function formatOpenDocumentUrl(document) {
  return '/documents/' + document.slug + '/' + document.title.replace(/\s/g, '-')
  // return getUrlFormat('/file_viewer/', {
  //   'document_id': documentId,
  //   'csrfmiddlewaretoken': getCookie('csrftokean'),
  // })
}


function getValFromUrlParam(key) {
  return new URL(window.location.href).searchParams.get(key)
}


function copyToClipboard(content) {
  const temp = document.createElement('input')
  temp.style.hidden = true
  document.body.appendChild(temp)
  temp.value = content
  temp.select()
  document.execCommand('copy')
  document.body.removeChild(temp)
}


String.prototype.format = function() {
  let a = this
  for (let k in arguments) {
    a = a.replace('{' + k + '}', arguments[k])
  }
  return a
}

function renderMathJax() {
  if (window.hasOwnProperty('MathJax'))
    MathJax.Hub.Queue(['Typeset', MathJax.Hub])
}

export {
  getCookie,
  copyToClipboard,
  getUrlFormat,
  hexToRgb,
  formatOpenCoterieDocumentUrl,
  formatOpenDocumentUrl,
  getValFromUrlParam,
  renderMathJax
}




