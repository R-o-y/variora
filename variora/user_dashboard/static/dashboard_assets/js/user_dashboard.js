'use strict';

function addEditDocTitleListener() {
    // change document title
    $(".EditDocTitleButton").on("click", function() {
        var $td = $(this).parents("td");
        var origDocTitle = $td.find("span").text();
        $td.html("<input type='text'></input>&nbsp<i class='fa fa-check-circle' style='cursor: pointer' aria-hidden='true'></i>");
        if (userDashboardPageType == "administrated_coterie_page")
            $td.find("input").css("width", String($td.width() - $td.find("i").width() - 80) + "px");
        else
            $td.find("input").css("width", String($td.width() - $td.find("i").width() - 28) + "px");
        $td.find("input").val(origDocTitle);

        $td.find("i").on("click", function() {
            var new_doc_title = $td.find("input").val();
            if (new_doc_title != origDocTitle) {

                if (userDashboardPageType == "documents_page")
                    var action = "/file_viewer/edit_doc_title";
                else if (userDashboardPageType == "administrated_coterie_page")
                    var action = "/coterie/edit_coteriedoc_title";

                $.ajax({
                    type: "POST",
                    url: action,
                    data: {
                        csrfmiddlewaretoken: getCookie('csrftoken'),
                        document_id: $td.parents("tr").find("input[name='document_id']").val(),
                        new_doc_title: new_doc_title,
                    },
                });
            }
            $td.html('<i class="fa fa-pencil-square-o EditDocTitleButton" style="cursor: pointer" aria-hidden="true"></i>&nbsp' + '<span>' + new_doc_title + '</span>');
            addEditDocTitleListener();
        });
    });
}

$(document).ready(function() {
    $(".pe-7s-plus").on("click", function() {
        var create_group_form_layer = layer.open({
            type: 1,
            title: "Create a new group",
            skin: 'layui-layer-demo',
            closeBtn: 1,
            shift: 4,
            area: ['380px', '280px'],
            shadeClose: true, //开启遮罩关闭
            content: '\
                <form id="create_group_form" style="margin-left: auto; margin-right: auto; margin-top: 28px; width: 200px;">\
                    <input name="coterie_name" type="text" class="form-control" placeholder="name"><br>\
                    <textarea name="coterie_description" type="text" class="form-control" rows="2" placeholder="description"></textarea><br>\
                    <button class="btn btn-info" type="button" style="float: right;">create</button>\
                </form>\
            '
        });

        $("#create_group_form").find("button").on("click", function() {
            var $form = $("#create_group_form");
            $(this).css("disabled", "true");
            $.ajax({
                type: "POST",
                url: "/coterie/handle_create_coterie",
                data: {
                    csrfmiddlewaretoken: getCookie('csrftoken'),
                    coterie_name: $form.find("input").val(),
                    coterie_description: $form.find("textarea").val(),
                },
                success: function () {
                    layer.close(create_group_form_layer);
                    window.location.reload();
                },
            });
        });
    });

    // add indexes for table
    $("tbody").each(function() {
        var $tbody = $(this);
        var length = $tbody.children("tr").length;     
        for (var i = 0; i < length; i++) {
            $($tbody.children("tr")[i]).children("td:first").text(i + 1);
        }
    });

    // confirmation after clicking delete
    $(".FileDeleteBtn").on("click", function() {
        var thisBtn = $(this);
        layer.confirm('confirm delete?', 
            {
                btn: ['yes'], //按钮
                title: false,
                shadeClose: true, //开启遮罩关闭
            }, 
            function() {
                layer.msg('delete successfully', {icon: 1});
                $.ajax({
                    type: "DELETE", url: thisBtn.attr("action-link"),
                    beforeSend: function(xhr) { xhr.setRequestHeader("X-CSRFToken", getCookie("csrftoken")); },
                    success: function () { window.location.reload(); },
                });
            }
        );
    });
    // confirmation after clicking remove member
    $(".MemberRemoveForm").find("button").on("click", function() {
        var thisBtn = $(this);
        layer.confirm('confirm remove this member?', 
            {
                btn: ['yes'], //按钮
                title: false,
                shadeClose: true, //开启遮罩关闭
            }, 
            function() {
                layer.msg('remove successfully', {icon: 1});
                thisBtn.parents(".MemberRemoveForm").submit();
            }
        );
    });

    addEditDocTitleListener();
});

function getCookie(name) {
    var cookieValue = null;
    if (document.cookie && document.cookie != '') {
        var cookies = document.cookie.split(';');
        for (var i = 0; i < cookies.length; i++) {
            var cookie = jQuery.trim(cookies[i]);
            // Does this cookie string begin with the name we want?
            if (cookie.substring(0, name.length + 1) == (name + '=')) {
                cookieValue = decodeURIComponent(cookie.substring(name.length + 1));
                break;
            }
        }
    }
    return cookieValue;
}
