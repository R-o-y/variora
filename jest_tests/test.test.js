import { getCookie, getUrlFormat, getValFromUrlParam, groupAvatarColors } from 'util.js'
import { validateDocumentTitle } from 'home_util.js'


test('adds 1 + 2 to equal 3', () => {
  expect(1 + 2).toBe(3)
})


test('test validateDocumentTitle', () => {
  var result = validateDocumentTitle('')
  expect(result).toEqual(false)
})


test('test getUrlFormat', () => {
  var params = {
    id: '1618577571113001418',
    wfr: 'spider',
    for: 'pc'
  }
  var urlBase = 'https://baijiahao.baidu.com/s'
  var result = getUrlFormat(urlBase, params)
  expect(result).toEqual('https://baijiahao.baidu.com/s?id=1618577571113001418&wfr=spider&for=pc&')
})


test('test string format', () => {
  var result = 'test {0} format'.format('string')
  expect(result).toEqual('test string format')

  var result2 = 'test {0} {1}'.format('string', 'format')
  expect(result2).toEqual('test string format')
})
