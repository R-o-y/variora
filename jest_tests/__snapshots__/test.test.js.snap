// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test react 1`] = `
<div
  className="ant-layout-content"
  style={
    Object {
      "margin": 0,
      "minHeight": 280,
      "paddingLeft": 18,
      "paddingRight": 18,
      "paddingTop": 16,
    }
  }
>
  <ul
    aria-activedescendant=""
    className="ant-menu ant-menu-horizontal card ant-menu-light ant-menu-root"
    onKeyDown={[Function]}
    role="menu"
    style={
      Object {
        "padding": 0,
      }
    }
    tabIndex="0"
  >
    <li
      aria-selected={true}
      className="ant-menu-item-selected ant-menu-item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="anticon anticon-file"
        />
        Uploaded Documents
      </a>
    </li>
    <li
      aria-selected={false}
      className="ant-menu-item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={Object {}}
    >
      <a
        href="/collected"
        onClick={[Function]}
      >
        <i
          className="anticon anticon-heart-o"
        />
        Collected Documents
      </a>
    </li>
  </ul>
  <div>
    <div
      className="card"
      style={
        Object {
          "marginTop": 16,
          "overflow": "auto",
        }
      }
    >
      <div
        className="ant-table-wrapper"
      >
        <div
          className="ant-spin-nested-loading"
          style={null}
        >
          <div
            className="ant-spin-container"
          >
            <div
              className="ant-table ant-table-large ant-table-empty ant-table-scroll-position-left"
              style={Object {}}
            >
              <div
                className="ant-table-content"
              >
                <div
                  className="ant-table-body"
                  onScroll={[Function]}
                  style={Object {}}
                >
                  <table
                    className=""
                    style={Object {}}
                  >
                    <colgroup>
                      <col
                        style={
                          Object {
                            "minWidth": "20%",
                            "width": "20%",
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": "40%",
                            "width": "40%",
                          }
                        }
                      />
                      <col
                        style={
                          Object {
                            "minWidth": "40%",
                            "width": "40%",
                          }
                        }
                      />
                    </colgroup>
                    <thead
                      className="ant-table-thead"
                    >
                      <tr
                        style={null}
                      >
                        <th
                          className=""
                        >
                          <span>
                            Id
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span>
                            Title
                          </span>
                        </th>
                        <th
                          className=""
                        >
                          <span>
                            Action
                          </span>
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      className="ant-table-tbody"
                    />
                  </table>
                </div>
                <div
                  className="ant-table-placeholder"
                >
                  <span>
                    <i
                      className="anticon anticon-frown-o"
                    />
                    暂无数据
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card"
      style={
        Object {
          "marginTop": 16,
          "overflow": "auto",
          "padding": 18,
        }
      }
    >
      <div
        className="ant-row"
      >
        <div
          className="ant-col-12"
          style={
            Object {
              "textAlign": "left",
            }
          }
        >
          <span
            className=""
          >
            <div
              className="ant-upload ant-upload-select ant-upload-select-text"
              style={
                Object {
                  "display": "",
                }
              }
            >
              <span
                className="ant-upload"
                onClick={[Function]}
                onDragOver={[Function]}
                onDrop={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex="0"
              >
                <input
                  accept="application/pdf"
                  multiple={false}
                  onChange={[Function]}
                  style={
                    Object {
                      "display": "none",
                    }
                  }
                  type="file"
                />
                <button
                  className="ant-btn"
                  onClick={[Function]}
                  style={
                    Object {
                      "margin": 8,
                    }
                  }
                  type="button"
                >
                  <i
                    className="anticon anticon-file-add"
                  />
                  <span>
                     Click to Choose Local File
                  </span>
                </button>
              </span>
            </div>
            <div
              className="ant-upload-list ant-upload-list-text"
            />
          </span>
          <input
            className="ant-input"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Give a title to this document"
            style={
              Object {
                "margin": 8,
                "width": "60%",
              }
            }
            type="text"
            value=""
          />
          <div>
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              style={
                Object {
                  "margin": 8,
                }
              }
              type="button"
            >
              <i
                className="anticon anticon-upload"
              />
              <span>
                upload
              </span>
            </button>
          </div>
        </div>
        <div
          className="ant-col-12"
          style={
            Object {
              "textAlign": "left",
            }
          }
        >
          <input
            className="ant-input"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="URL to an online document"
            style={
              Object {
                "margin": 8,
                "width": "60%",
              }
            }
            type="text"
            value=""
          />
          <input
            className="ant-input"
            disabled={false}
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder="Give a title to this document"
            style={
              Object {
                "margin": 8,
                "width": "60%",
              }
            }
            type="text"
            value=""
          />
          <div>
            <button
              className="ant-btn ant-btn-primary"
              onClick={[Function]}
              style={
                Object {
                  "margin": 8,
                }
              }
              type="button"
            >
              <i
                className="anticon anticon-upload"
              />
              <span>
                upload
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
